include/unifex/transform_stream.hpp

```diff
@@ -27,7 +27,7 @@ namespace _tfx_stream {
     auto operator()(StreamSender&& stream, Func&& func) const {
       return next_adapt_stream(
           (StreamSender &&) stream, [func = (Func &&) func](auto&& sender) mutable {
-            return transform((decltype(sender))sender, std::ref(func));
+            return transform((decltype(sender))sender, (Func &&)func);
```

Comment:
The use of `std::ref()` here was intentional to allow the use of stateful function objects.

Also, the function will be called once for each `next()` and so we shouldn't be moving the function into the transform-sender.

Reply:
this appears to work now - I have reverted the change